src/Tools/dotnet-monitor/Logs/LogsOperation.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Monitoring.EventPipe;
+using Microsoft.Diagnostics.Monitoring.Options;
+using Microsoft.Diagnostics.Monitoring.WebApi;
+using Microsoft.Diagnostics.NETCore.Client;
+using Microsoft.Extensions.Logging;
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Utils = Microsoft.Diagnostics.Monitoring.WebApi.Utilities;
+
+namespace Microsoft.Diagnostics.Tools.Monitor
+{
+    internal sealed class LogsOperation : IArtifactOperation
+    {
+        private readonly IEndpointInfo _endpointInfo;
```

Comment:
Consider adding an abstract base class that inherits from IArtifactOperation and contains some of the shared fields and shared logic across all artifact types.